### PR TITLE
libsdl2: use libsdl2-powerpc on powerpc systems (no change for other platforms)

### PR DESCRIPTION
**Pull request by barracuda156**
[skip ci]

#### Description

There is no change to the code, just a switch to path-style dependency.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by mohd-akram** [comment]
What's the reason for it?

**Comment by barracuda156** [comment]
@mohd-akram A [port](https://ports.macports.org/port/libsdl2-powerpc) of SDL2 for powerpc.

(Alternatively, `libsdl2` port can be modified to use `libsdl2-powerpc` on powerpc systems, but I am not sure if that is preferable and if Josh will be okay with that change. Path-style solution on the other hand is a standard practice for such cases.)

**Review by mohd-akram — APPROVED** [review]

**Review by i0ntempest — APPROVED** [review]

**Comment by jmroot** [comment]
> (Alternatively, `libsdl2` port can be modified to use `libsdl2-powerpc` on powerpc systems, but I am not sure if that is preferable and if Josh will be okay with that change.

Yes that's preferable to modifying everything else.

**Comment by barracuda156** [comment]
> > (Alternatively, `libsdl2` port can be modified to use `libsdl2-powerpc` on powerpc systems, but I am not sure if that is preferable and if Josh will be okay with that change.
> 
> Yes that's preferable to modifying everything else.

@jmroot Sure, making it a draft now and gonna drop path-style commit and make a modification to `libsdl2` port.

**Comment by barracuda156** [comment]
@jmroot Could you please take a look if this is correct now?

**Comment by barracuda156** [comment]
@i0ntempest Yes, I dropped per-port changes, following recommendation of @jmroot above. If that is merged, we do not need path-style dependencies.

File: devel/libsdl2/Portfile [hunk 1]
```diff
@@ -21,8 +21,9 @@ homepage        https://www.libsdl.org/
 
 github.tarball_from releases
 distname        SDL2-${version}
-checksums       rmd160 9a8b7d06bc57b64c62aad1addafbd98d57edddda \
-                sha256 01215ffbc8cfc4ad165ba7573750f15ddda1f971d5a66e9dcaffd37c587f473a
+checksums       rmd160  9a8b7d06bc57b64c62aad1addafbd98d57edddda \
+                sha256  01215ffbc8cfc4ad165ba7573750f15ddda1f971d5a66e9dcaffd37c587f473a \
+                size    7428023
```
[the remark this answers is not in the document]
Comment (barracuda156) [reply devel/libsdl2/Portfile:26]: Got it, will drop this now.

File: devel/libsdl2/Portfile [hunk 2]
```diff
@@ -32,7 +33,17 @@ configure.args  --without-x \
...
+if {${os.subplatform} eq "macosx" && (${configure.build_arch} in [list ppc ppc64])} {
+    depends_lib   port:libsdl2-powerpc
+    distfiles
+    patchfiles
+    use_configure no
+    build {}
+    destroot {
+        xinstall -d ${destroot}${prefix}/share/doc/${name}
+        system "echo $name is a stub port > ${destroot}${prefix}/share/doc/${name}/README"
+    }
```
[the remark this answers is not in the document]
Comment (barracuda156) [reply devel/libsdl2/Portfile:40]: Done, and dropped a commit with size.

**Comment by barracuda156** [comment]
> And 10.4 and 10.5 Intel?

@kencu This is perhaps not something which I am good with (besides of not having a setup for those on x86).

Given that we should use GCC on any PowerPC, while on x86 the preference is towards Clangs, I think it makes little sense to mix these into a single port: it will make code more complicated and maintaining it a mess. (And as you well know with ObjC it matters substantially.)

Perhaps, `libsdl2-snowleopard` could be eventually made into `-legacy` and support all early x86 systems?

**Comment by kencu** [comment]
well -- if this works, it should probably be used on all systems < 10.6.

but then, we have that 10.6 on PPC issue again ...

**Review by jmroot — APPROVED** [review]

**Comment by jmroot** [comment]
> well -- if this works, it should probably be used on all systems < 10.6. It is an SDK problem, not really an architecture problem.

If you've confirmed that it works on 10.4/10.5 Intel, I certainly have no objection to renaming `libsdl2-powerpc` to something more appropriate and changing the condition in libsdl2 to `${os.major} < 10 || ${configure.build_arch} in [list ppc ppc64]`.

**Comment by kencu** [comment]
that will take some time, and no need to hold this up any longer.

**Pull request merged**